### PR TITLE
Load resources no matter how dartdoc is run

### PR DESCRIPTION
**Pull request by sethladd**

### CODE REVIEW

File: bin/dart_resource_loader.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import 'package:dartdoc/resource_loader.dart';
+
+main() async {
+  print(await loadAsString('package:dartdoc/resources/styles.css'));
```
Comment (devoncarew) [inline bin/dart_resource_loader.dart]: +1, awesome!

File: pubspec.yaml [hunk 2]
```diff
@@ -17,6 +17,9 @@ dependencies:
   path: '^1.3.0'
   quiver: '>=0.18.0 <0.22.0'
   yaml: '^2.1.0'
+  pub_cache:
```
Comment (sethladd) [inline pubspec.yaml]: Thanks. Fixed.

File: bin/dartdoc.dart [hunk 3]
```diff
@@ -54,6 +56,16 @@ void main(List<String> arguments) {
     ..generateDocs();
 }
 
+exitWhenRunFromSnapshot() {
+  var script = Platform.script;
+  print(script);
+  if (script.toString().contains('snapshot')) {
```
Comment (sethladd) [inline bin/dartdoc.dart]: The intention of this PR changed. We will support running in a globally activated snapshot. So I'll remove this :)

**Comment by sethladd** [comment]
Also addresses https://github.com/dart-lang/dartdoc/issues/370

**Comment by sethladd** [comment]
I'm going to merge this and do further testing after pushing to pub.

**Pull request merged**